posthog/models/event/sql.py

```diff
@@ -303,11 +303,38 @@
     team_id,
     distinct_id,
     elements_chain,
-    created_at
+    created_at,
+    session_recording_events.session_id as session_id,
```

Comment:
I was curious why we need the `LEFT JOIN` at all, when all we're taking from it is the `session_id` that already exists on the event?

I'm probably missing something here, but it _appears_ to me right now that we're just slowing down the event query here for no gain?

Reply:
(cc: @benjackwhite @alexkim205) - tagging so you see this :)

Reply:
The thinking here was not to show the recording button if no recording exists (due to falling past the TTL or ingestion issues etc.)

You're right it does seem that events are much slower. Tempted to revert this and go for simply using the existence of certain properties to decide whether to show the button or not

Reply:
Refreshing my mind on this topic - I don't think this is easy to get around. If we don't do the join check then we can never really be sure. There are too many things that can lead to a missing recording:
1. Recordings being disabled at the time of ingestion
2. TTL pushing recording out of the history
3. Recording ingestion delays
4. etc.